### DISCUSSION TITLE
Make back links optional

### DISCUSSION THREAD

**Discussion by Adanufgail** [thesamim/TickTickSync#192] [Polls]
Per [Issue 55](https://github.com/thesamim/TickTickSync/issues/55),

**Comment by Adanufgail** [comment]
Because you require a the ticktick id, you could add it as a hashtag in Obsidian with a customizable prefix as an option to prevent collisions. Hashtags are hidden from view in Tasks. 

And then just not show the back link on the Tick Tick side. It really clutters the UI and makes it difficult at a glance to parse notifications

**Reply by johnyoonh** [reply]
Yeah, the view from Ticktick UI is pretty cluttered with those back links

**Reply by on5ap** [reply]
I was also totally overwhelmed by the amount of space taken by all those redundant backlinks, added to title of every, and all referring to the same "[Path/Inbox.md] ...
In the TickTick app the name of the viewed list is marked at the top of the screen, so no need to remind it with every task.

Off course an option to add/remove those backlinks would be the best way ... ;-))
But maybe, an icon, a symbol or just one single character as an indicator of the backlink instead of having [Path/Listname.md] in the title of each task, would help a lot :-)

**Comment by thesamim** [comment]
@Adanufgail , @johnyoonh , @on5ap : Link location is now configurable. 

Let me know your thoughts.

**Comment by on5ap** [comment]
YES! I like the Link Location option !
(I also liked my suggestion to have an indication of the link without the path being visual in the title) 

Euh, I sounded good, but there is maybe something wrong.

So I chose "No Link" for both Obsidian and TickTick.
After manual Sync, nothing happend, links where still there.

So I started to remove some links in the task titles from one list "SkyLane" from within TickTick.
At the next Sync several duplicates where found in 2 lists : the list "SkyLane.md" where I removed the links, but also in the Inbox.md
The tasks from "SkyLane.md" are also found in the "Inbox.md"

TickTickSync reports :
Please fix manually to avoid
unpredictable results.
Sync is aborted to prevent data corruption.

Every Sync ends with the same error.

Maybe the best thing would be to restart all over again.

What is the best way to start all over ?  
Remove TickTickSync plugin from vault
Remove all related TickTickSync files  
Reinstall TickTickSync from scratch 

If you need other info, please let me know.

F

**Reply by thesamim** [reply]
The links settings are applied on the next update of the task. As in: When you updated a task in TT, if you chose "No Links" on the next sync, the link will be removed from Obsidian. When a task is updated in OBS, if you chose "No Links" on the next sync, the link will be removed from  TT.

I chickened out from creating an automatic way of resyncing everything all at once because of the potential confusion it would cause. It was easy enough for the Notes delimeter configuration because it's a sync from TT to Obsidian only. (And you might want to give that a shot by the way, it's under the "Notes" tab in settings.  It will force everything to update from TT to Obsidian. Going the other way is a bit more problematic.

>So I started to remove some links in the task titles from one list "SkyLane" from within TickTick.
At the next Sync several duplicates where found in 2 lists : the list "SkyLane.md" where I removed the links, but also in the Inbox.md
The tasks from "SkyLane.md" are also found in the "Inbox.md"

That scares me. I thought I had taken care of the use cases that would cause duplicates. Just to make sure; you're on 1.1.10 right? If you are, Please tell me what your settings are in data.json for:
```
  "enableFullVaultSync",
  "tagAndOr",
  "debugMode",
  "logLevel",
  "SyncProject",
  "SyncTag",
  "defaultProjectName",
```

**Reply by on5ap** [reply]
Yes I'm in 1.1.10. ;-)

Here are the last settings from data.json

  "enableFullVaultSync": false,
  "tagAndOr": 1,
  "debugMode": false,
  "logLevel": "info",
  "SyncProject": "",
  "SyncTag": "",
  "defaultProjectId": "",
  "defaultProjectName": "Inbox",
  "TickTickTasksFilePath": "T",
  "keepProjectFolders": false,
  "syncNotes": false,
  
As you can see, I reduced the FilePath to just "T" to reduce length of the path in the title to the minimum ;-)

To stop the error messages and / prevent further damage, I now deactivated the plugin, 
but I haven't yet removed it from my Vault, so if you want more info, just let me know, as I want to reinstall the link from scratch.

**Comment by on5ap** [comment]
I see that version 1.1.11 is available.
but installing it is not possible, I get the failed to install plugin message

I still have 1.1.10 in my vault but not enabled, 
enabling it, does not work either

**Reply by thesamim** [reply]
Sorry, that was premature. 1.1.11-beta is available for testing, not 1.1.11 proper. 

If you'd like to test please see:
https://github.com/thesamim/TickTickSync/issues/273#issuecomment-3103891691 for installation instructions.

**Comment by johnyoonh** [comment]
@thesamim I tried the latest. The no link and no tags did not work when I did a vault vault sync

**Comment by johnyoonh** [comment]
@thesamim I also tried without full vault sync and the ticktick tag still shows up in Obsidian side

**Reply by thesamim** [reply]
Ya, sorry, the only way to get the change to apply is to actually update the tasks.

I have not found a safe way to force that update.

IF all your tasks are up to date on ticktick, and you are feeling brave : go to the "Notes" tab in settings and click on "Reset Notes" and that will reset the Obsidian tasks to respect you links settings.

There is not a corresponding way to reset the ticktick side, but if you are feeling brave, you can globally add a tag to your tasks, force the sync and then finally remove that tag.

eg:
1. Using your favorite method globally edit "#ticktick" to "#ticktick #updateAllTasksNow"
2. Force a sync, wait patiently
3. When done reverse this by Using globally editing "#ticktick #updateAllTasksNow" to "#ticktick" 

Disclaimer: IF you have tasks in files that are not associated with default projects, AND those tasks have children: There's an edge case where the parentage could be lost (it depends on what default project the tasks are going to go to in TickTick.)

**Reply by johnyoonh** [reply]
I am okay with changes being applied only when task is changed. I think it makes sense as this a long operation. On Ticktick side, It should be simple for most people to go to `#ticktick` tag page, select all tags, remove that `#ticktick` tag from all tasks, and everything should sync going forward too

**Reply by johnyoonh** [reply]
But this still doesn't address `#ticktick` tag being added on Obsidian side when a new task is added (with full vault enabled)
or new task not getting created (without full vault enabled) unless putting `#ticktick` tag to trigger the add. Tasks should get added  automatically when "Default Project" or "Limit Sync to Project" without `#ticktick` tag

**Reply by thesamim** [reply]
> But this still doesn't address `#ticktick` tag being added on Obsidian side when a new task is added (with full vault enabled) or new task not getting created (without full vault enabled) unless putting `#ticktick` tag to trigger the add. Tasks should get added automatically when "Default Project" or "Limit Sync to Project" without `#ticktick` tag

Please open a new issue for this.

**Reply by on5ap** [reply]
As I do not want all Obsidian tasks to appear in TT, I'm ok with the #ticktick tags on Obsidian side, they remind me that the task has being transferred and can also to be managed in/from TT ...

**Comment by johnyoonh** [comment]
I wish I could limit the full sync to to default folder location instead of relying on tags or project. ticktick_id should be sufficient.

**Reply by thesamim** [reply]
1. Using TickTick ID alone is fine for updates. Dealing with Adds, if "Full Vault Sync" is not turned on AND a "Default Project" or "Limit Sync to Project" is not set, is problematic.
2. Regardless: If it's important enough, please open new issue (prefix the title with [FR] please) and if it gets enough support I will consider ways to get around that....
